src/kili/domain/annotation.py

```diff
@@ -0,0 +1,126 @@
+"""Annotation domain."""
+
+from typing import List, Literal, NewType, Optional, TypedDict, Union
+
+from .label import LabelId
+from .ontology import JobName
+
+AnnotationId = NewType("AnnotationId", str)
+KeyAnnotationId = NewType("KeyAnnotationId", str)
+
+
+class Vertice(TypedDict):
+    """Vertice."""
+
+    x: float
+    y: float
+
+
+class ObjectDetectionAnnotationValue(TypedDict):
+    """Object detection annotation value."""
+
+    vertices: List[List[List[Vertice]]]
+
+
+class ClassificationAnnotationValue(TypedDict):
+    """Classification annotation value."""
+
+    categories: List[str]
+
+
+class TranscriptionAnnotationValue(TypedDict):
+    """Transcription annotation value."""
+
+    text: str
+
+
+class Annotation(TypedDict):
+    """Annotation."""
+
+    id: AnnotationId
+    labelId: LabelId
+    job: JobName
+    path: List[List[str]]
+
+
+class FrameInterval(TypedDict):
+    """Frame interval."""
+
+    start: int
+    end: int
+
+
+class VideoObjectDetectionKeyAnnotation(TypedDict):
+    """Video object detection key annotation."""
+
+    id: KeyAnnotationId
+    isPrediction: bool
+    frame: int
+    annotationValue: ObjectDetectionAnnotationValue
+
+
+class VideoClassificationKeyAnnotation(TypedDict):
+    """Video classification key annotation."""
+
+    id: KeyAnnotationId
+    isPrediction: bool
+    frame: int
+    annotationValue: ClassificationAnnotationValue
+
+
+class VideoTranscriptionKeyAnnotation(TypedDict):
+    """Video transcription key annotation."""
+
+    id: KeyAnnotationId
+    isPrediction: bool
```

Comment:
Care, as `isPrediction` is moving to annotationValue : https://gitlab.com/kili-technology/kili/-/merge_requests/11110